scalding-core/src/main/scala/com/twitter/scalding/TupleConverter.scala

```diff
@@ -51,6 +53,37 @@ trait LowPriorityTupleConverters extends java.io.Serializable {
     }
 }
 
+/**
+ * Based on `FromTraversable` from shapeless
+ */
+trait FromTupleEntry[Out <: HList, I <: Nat] {
+  def apply(l : TupleEntry) : Option[Out]
+}
+
+object FromTupleEntry {
+  def apply[Out <: HList](implicit from: FromTupleEntry[Out, Nat._0]) = from
+
+  implicit def hnilFromTupleEntry[T, I <: Nat](implicit toIntI : ToInt[I]) =
+    new FromTupleEntry[HNil, I] {
+      def apply(te : TupleEntry) =
+        if(te.size() == toIntI()) Some(HNil) else None
```

Comment:
shouldn't we run `toInt` once before creating the instance to guarantee that is only run once? We definitely don't work to run that on each tuple.